### PR TITLE
zoltan: Add PGI runtime libs to LDFLAGS

### PR DESCRIPTION
**Pull request by quellyn**
`zoltan` fails to compile with PGI 19.7 unless provided with the PGI runtime libraries. This is similar to an old [Nvidia forum issue](https://forums.developer.nvidia.com/t/petsc-on-linux86-64/130306)

This patch adds PGI runtime libraries to LDFLAGS when '%pgi' is the case.

### CODE REVIEW

File: var/spack/repos/builtin/packages/zoltan/package.py [hunk 1]
```diff
@@ -90,6 +90,13 @@ def configure_args(self):
             '-g' if '+debug' in spec else '',
         ]
 
+        config_ldflags = []
+        # PGI runtime libraries
+        if '%pgi' in spec:
+            config_ldflags.append('-pgf90libs')
+        if '+parmetis' in spec:
+            config_ldflags.append('-L{0}'.format(spec['metis'].prefix.lib))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/zoltan/package.py]: Why was this line moved up here?

Comment (quellyn) [reply var/spack/repos/builtin/packages/zoltan/package.py]: Hi Adam,

> Why was this line moved up here?

I saw that there were list variables for both `config_args` and `config_cflags` and thought I should follow suite with `config_ldflags`. Then I saw that lone `--with-ldflags` switch for `parmetis` down below and thought that perhaps it was more consistent to treat all those together. I'm happy to retract that bit; I just got carried away.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/zoltan/package.py]: I would probably undo that change if the previous way works

Comment (quellyn) [reply var/spack/repos/builtin/packages/zoltan/package.py]: Okay, reverted the parmetis change. Now the patch just defines `config_ldflags` and applies the fix for PGI.

**Review by adamjstewart — APPROVED** [review]
